### DISCUSSION TITLE
watch()

### DISCUSSION THREAD

**Discussion by mjcarnaje** [react-hook-form/react-hook-form#3520] [Q&A] [answered]
how to use watch that avoid re-rendering? I use switch(checkbox) to watch and it is a property of and object inside of an array, and it causes lag because the component re render when i type

**Comment by mjcarnaje** [comment, marked as answer]
> how to use watch that avoid re-rendering? I use switch(checkbox) to watch and it is a property of and object inside of an array, and it causes lag because the component re render when i type
ohh never mind, I figured it out hahaha

**Reply by franciscocobas** [reply]
So, @mjcarnaje did you find the answer?

**Reply by mjcarnaje** [reply]
sorry sir i already forgot about it hahah but you can see the code on how to implement it, but i guess you already figure it out because it's been 8 days since you 've commented here.

**Reply by spoeken** [reply]
How did you fix this?
